app/api/v1/farmers.py

```
@@ -0,0 +1,88 @@
+):
+    """Update farmer profile."""
+    farmer_service = FarmerService(db)
+    farmer = await farmer_service.update_farmer_profile(farmer_id, farmer_data)
+    if not farmer:
+        raise HTTPException(status_code=404, detail="Farmer not found")
+    return farmer
+
+@router.delete("/{farmer_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_farmer(
+    farmer_id: int,
+    db: AsyncSession = Depends(get_db)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)